DQM/L1TMonitor/src/L1TCaloLayer1Summary.cc

```diff
@@ -0,0 +1,122 @@
+    etMatrix[1][sRegion.gctEta()][sRegion.gctPhi()] = sRegion.et();
+  }
+  int iRegion = 0;
+  for (int iEta = 0; iEta < 18; iEta++) {
+    for (int iPhi = 0; iPhi < 18; iPhi++) {
+      if (foundMatrix[0][iEta][iPhi] && foundMatrix[1][iEta][iPhi]) {
+        histoCaloRegions->Fill(iRegion, etMatrix[0][iEta][iPhi]);
+        histoSimRegions->Fill(iRegion, etMatrix[1][iEta][iPhi]);
+        histoCaloMinusSimRegions->Fill(iRegion, etMatrix[0][iEta][iPhi] - etMatrix[1][iEta][iPhi]);
+        iRegion++;
+      }
+    }
+  }
+
+  float caloCICADAScore = iEvent.get(caloLayer1CICADAScoreToken_)[0];
+  auto gtCICADAScores = iEvent.get(gtCICADAScoreToken_);
+  float simCICADAScore = iEvent.get(simCICADAScoreToken_)[0];
+
+  histoSimCICADAScore->Fill(simCICADAScore);
+  histoCaloMinusSim->Fill(caloCICADAScore - simCICADAScore);
+
+  uint32_t bx0Idx;
+  if (gtCICADAScores.size() == 30) {
+    bx0Idx = 12;
+  } else if (gtCICADAScores.size() == 5) {
+    bx0Idx = 2;
+  } else {
+    bx0Idx = 2;
+  }
+
+  histoCaloLayer1CICADAScore->Fill(caloCICADAScore);
+  histoGtCICADAScore->Fill(gtCICADAScores[bx0Idx]);
+  histoCaloMinusGt->Fill(gtCICADAScores[bx0Idx] - caloCICADAScore);
+}
+
+void L1TCaloLayer1Summary::bookHistograms(DQMStore::IBooker& ibooker, edm::Run const&, edm::EventSetup const&) {
+  ibooker.setCurrentFolder(histFolder_);
+  histoSlot7MinusDaqBxid = ibooker.book1D("slot7BXID", "Slot 7- DAQ BXID", 50, -20, 20);
+
+  ibooker.setCurrentFolder(histFolder_ + "/CICADAScore");
+  histoCaloLayer1CICADAScore = ibooker.book1D("caloLayer1CICADAScore", "CaloLayer1 CICADAScore", 50, 0, 200);
+  histoGtCICADAScore = ibooker.book1D("gtCICADAScore", "GT CICADAScore at BX0", 50, 0, 200);
+  histoCaloMinusGt = ibooker.book1D("caloMinusGtCICADAScore", "CaloLayer1 - GT CICADAScore at BX0", 50, -50, 50);
+  histoSimCICADAScore = ibooker.book1D("simCaloLayer1CICADAScore", "simCaloLayer1 CICADAScore", 50, 0, 200);
+  histoCaloMinusSim = ibooker.book1D("caloMinusSimCICADAScore", "CaloLayer1 - simCaloLayer1 CICADAScore", 50, -50, 50);
+
+  ibooker.setCurrentFolder(histFolder_ + "/Regions");
+  histoCaloMinusSimRegions = ibooker.book2D(
+      "caloMinusSumRegions", "CaloLayer1 - simCaloLayer1 Regions;Region;ET", 252, -0.5, 252.5, 100, -400, 400);
+  histoCaloRegions = ibooker.book2D("caloLayer1Regions", "CaloLayer1 Regions;Region;ET", 252, -0.5, 252.5, 100, 0, 800);
+  histoSimRegions =
+      ibooker.book2D("simCaloLayer1Regions", "simCaloLayer1 Regions;Region;ET", 252, -0.5, 252.5, 100, 0, 800);
+}
+
+// ------------ method fills 'descriptions' with the allowed parameters for the module  ------------
+void L1TCaloLayer1Summary::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  //The following says we do not know what parameters are allowed so do no validation
+  // Please change this to state exactly what you do use, even if it is no parameters
+  edm::ParameterSetDescription desc;
+  desc.setUnknown();
+  descriptions.addDefault(desc);
```

Comment:
please provide a functioning `fillDescriptions`

Reply:
All of your comments have been accepted with changes implemented in the latest commit 182848e.

Reply:
thanks!